### PR TITLE
264 create the po

### PR DESCRIPTION
**Pull request by alishaevn**
# Story
we need to be able to create purchase orders from the webstore.

# Expected Behavior Before Changes
- we could not create a purchase order from the webstore

# Expected Behavior After Changes
- a purchase order is created for the respective sow when it's accepted

# Screenshots / Video

https://user-images.githubusercontent.com/29032869/228003043-75c291f6-9400-4d36-a6ad-62d7895ca0ac.mp4

# testing instructions
- create, or find, a webstore request with an sow
- accept the sow from the webstore
- make sure that when the page refreshes, the expected PO document is there

### CODE REVIEW

File: pages/requests/[uuid].js [hunk 1]
```diff
@@ -42,11 +42,14 @@ const Request = () => {
...
 
+  const [show, setShow] = useState(false)
+  const [message, setMessage] = useState(null)
   const [allPOs, setAllPOs] = useState([])
   const [isPOError, setIsPOError] = useState(false)
   const [isLoadingPOs, setIsLoadingPOs] = useState(true)
+
   useEffect(() => {
-    if (request) {
+    if (isLoadingPOs && request) {
```
Comment (alishaevn) [inline pages/requests/[uuid].js:52]: currently the `getAllPOs` function below is called nonstop for 2 reasons. this is the **second** reason.

every time the page is refreshed, `request` is still true so we are calling the api and "updating" state, which causes a rerender. (see attached video) 

https://user-images.githubusercontent.com/29032869/228001306-9da479e5-254a-4644-bb80-8e98dd2b9f8c.mp4

File: pages/requests/[uuid].js [hunk 2]
```diff
@@ -136,13 +173,9 @@ const Request = () => {
...
           {documents.length ? documents.map((document, index) => (
-            <Document 
+            <Document
               addClass='mt-3'
-              acceptSOW={acceptSOW(
-                request,
-                document.sowID,
-                accessToken,
-              )}
+              acceptSOW={() => handleAcceptingSOWandCreatingPO(document)}
```
Comment (alishaevn) [inline pages/requests/[uuid].js:178]: currently the `getAllPOs` function below is called nonstop for 2 reasons. this is the **first** reason.
the function was automatically being called, whether or not "accept sow" was pressed. (see attached video) 

https://user-images.githubusercontent.com/29032869/228001306-9da479e5-254a-4644-bb80-8e98dd2b9f8c.mp4

File: utils/api/requests.js [hunk 3]
```diff
@@ -22,8 +22,8 @@ export const useAllRequests = (accessToken) => {
   }
 }
 
-export const useOneRequest = (id, accessToken) => {
-  const { data, error } = useSWR(id ? [`/quote_groups/${id}.json`, accessToken] : null)
+export const useOneRequest = (uuid, accessToken) => {
+  const { data, error } = useSWR(uuid ? [`/quote_groups/${uuid}.json`, accessToken] : null)
```
Comment (alishaevn) [inline utils/api/requests.js:26]: better semantics.

File: pages/requests/[uuid].js [hunk 4]
```diff
@@ -55,9 +58,9 @@ const Request = () => {
         setIsPOError(isPOError)
       })()
     }
-  }, [allPOs, isPOError, request, uuid, session, accessToken])
+  }, [request, isLoadingPOs, accessToken, uuid])
 
-  const isLoading = isLoadingRequest || isLoadingSOWs || isLoadingFiles || isLoadingMessages || isLoadingPOs
```
Comment (alishaevn) [inline pages/requests/[uuid].js:60]: not checking for `isLoadingPOs` here allows us to close the sow modal and show the request. if this remained, closing the modal would hide the entire page behind a "loading" spinner.

**Comment by alishaevn** [comment]
this work was tested against acme-staging. it's ready for review, but will not merge until`beachsidebiotech` has been updated to `2.78.2`.

**Review by summer-cook — APPROVED** [review]

**Comment by alishaevn** [comment]
although `beachsidebiotech` hasn't been updated, I tested this code locally and nothing breaks. this is because although the po isn't being created on the prod api, the sow is still being accepted. since accepting an sow also returns a message, the flow created by this pr still works. the end result is just different.

once prod is updated, this code will allow the "accept sow" flow to _also_ create the purchase order.

**Pull request merged**